contracts/libs/Authenticity.sol

```diff
@@ -43,13 +48,17 @@ library Authenticity {
                 ),
                 "Authenticity: account does not exists"
             );
-
+            uint256 nonce = proof.states[i].nonce - 1;
+            // Add a huge value to avoid collision to the array initial value 0
+            uint256 safeIndex = _tx.fromIndex + 1000000000000000;
```

Comment:
The array `senders` is initialized with zeros. If we don't do this trick, we'll mistakenly subtract more nonces than we expect for senders with stateID 0.